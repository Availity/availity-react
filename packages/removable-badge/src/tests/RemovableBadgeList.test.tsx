import React from 'react';
import { render, waitFor, fireEvent } from '@testing-library/react';
import { BadgeItem, RemovableBadgeList } from '..';

const badgeList: BadgeItem[] = [
  { value: '1', color: 'primary', displayText: 'Test 1' },
  { value: '2', color: 'success', displayText: 'Test 2' },
  { value: '3', color: 'danger', displayText: 'Test 3' },
  { value: '4', color: 'warning', displayText: 'Test 4' },
  { value: '5', color: 'info', displayText: 'Test 5' },
  { value: '6', color: 'light', displayText: 'Test 6' },
  { value: '7', color: 'dark', displayText: 'Test 7' },
  { value: '8', displayText: 'Test 9' },
];

describe('Removable Badge List', () => {
  test('should render', async () => {
    const { container, getByTestId, getAllByTestId } = render(<RemovableBadgeList badges={badgeList} />);

    expect(container).toBeDefined();
    expect(container).toMatchSnapshot();

    const badgeListElement = await waitFor(() => getByTestId('removable_badge_list_section'));
    expect(badgeListElement).not.toBeNull();

    const badgeListItemElements = await waitFor(() => getAllByTestId('removable_badge_list_item'));
    expect(badgeListItemElements.length).toBe(badgeList.length);
  });

  test('should call onRemove onClick', async () => {
    const onRemove = jest.fn();

    const { container, getByTestId, getAllByTestId } = render(
      <RemovableBadgeList badges={badgeList} onRemove={onRemove} />
    );

    expect(container).toBeDefined();

    const badgeListElement = await waitFor(() => getByTestId('removable_badge_list_section'));
    expect(badgeListElement).not.toBeNull();

    const badgeListItemElements = await waitFor(() => getAllByTestId('removable_badge_list_item'));
    expect(badgeListItemElements).not.toBeNull();

    const firstBadgeElement = getAllByTestId('removable_badge_remove');
    fireEvent.click(firstBadgeElement[0]);

    await waitFor(() => {
      expect(onRemove).toHaveBeenCalledTimes(1);
    });
  });
});
