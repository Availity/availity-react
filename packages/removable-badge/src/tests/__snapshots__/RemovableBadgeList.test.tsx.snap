// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Removable Badge List should render 1`] = `
<div>
  <section
    data-testid="removable_badge_list_section"
  >
    <span
      class="ml-1 badge badge-primary"
      data-testid="removable_badge_list_item"
      id="removable_badge_list_1"
    >
      <i
        aria-hidden="true"
        class="icon icon-cancel"
        data-testid="removable_badge_remove"
        style="cursor: pointer;"
      />
      Test 1
    </span>
    <span
      class="ml-1 badge badge-success"
      data-testid="removable_badge_list_item"
      id="removable_badge_list_2"
    >
      <i
        aria-hidden="true"
        class="icon icon-cancel"
        data-testid="removable_badge_remove"
        style="cursor: pointer;"
      />
      Test 2
    </span>
    <span
      class="ml-1 badge badge-danger"
      data-testid="removable_badge_list_item"
      id="removable_badge_list_3"
    >
      <i
        aria-hidden="true"
        class="icon icon-cancel"
        data-testid="removable_badge_remove"
        style="cursor: pointer;"
      />
      Test 3
    </span>
    <span
      class="ml-1 badge badge-warning"
      data-testid="removable_badge_list_item"
      id="removable_badge_list_4"
    >
      <i
        aria-hidden="true"
        class="icon icon-cancel"
        data-testid="removable_badge_remove"
        style="cursor: pointer;"
      />
      Test 4
    </span>
    <span
      class="ml-1 badge badge-info"
      data-testid="removable_badge_list_item"
      id="removable_badge_list_5"
    >
      <i
        aria-hidden="true"
        class="icon icon-cancel"
        data-testid="removable_badge_remove"
        style="cursor: pointer;"
      />
      Test 5
    </span>
    <span
      class="ml-1 badge badge-light"
      data-testid="removable_badge_list_item"
      id="removable_badge_list_6"
    >
      <i
        aria-hidden="true"
        class="icon icon-cancel"
        data-testid="removable_badge_remove"
        style="cursor: pointer;"
      />
      Test 6
    </span>
    <span
      class="ml-1 badge badge-dark"
      data-testid="removable_badge_list_item"
      id="removable_badge_list_7"
    >
      <i
        aria-hidden="true"
        class="icon icon-cancel"
        data-testid="removable_badge_remove"
        style="cursor: pointer;"
      />
      Test 7
    </span>
    <span
      class="ml-1 badge badge-secondary"
      data-testid="removable_badge_list_item"
      id="removable_badge_list_8"
    >
      <i
        aria-hidden="true"
        class="icon icon-cancel"
        data-testid="removable_badge_remove"
        style="cursor: pointer;"
      />
      Test 9
    </span>
  </section>
</div>
`;
